### PR TITLE
Add a function FriendlyMissingBuffInRangeNotInCombatCheck

### PR DESCRIPTION
**Pull request by miraco**
Add EVENT_T_FRIENDLY_MISSING_BUFF_NOCOMBAT and rename EVENT_T_FRIENDLY_MISSING_BUFF to EVENT_T_FRIENDLY_MISSING_BUFF_INCOMBAT
Same for sd2 scripts

### Proof

- None

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

**Comment by AnonXS** [comment]
Have you tryed removing the combataction eventflag from this event so it is executed OOC?

**Comment by miraco** [comment]
That would not work. We have 2 combat checks here:
first one here
https://github.com/cmangos/mangos-tbc/blob/4fe8564c1cc213c4a0ac4214d3618900fc7e9be4/src/game/AI/EventAI/CreatureEventAI.cpp#L450
second one here
https://github.com/cmangos/mangos-tbc/blob/4fe8564c1cc213c4a0ac4214d3618900fc7e9be4/src/game/Grids/GridNotifiers.h#L880

And tbh i dont know if every creature that buffs friendly units do this out of combat

File: src/game/AI/EventAI/CreatureEventAI.h [hunk 1]
```diff
@@ -834,7 +835,8 @@ class CreatureEventAI : public CreatureAI
 
         bool SpawnedEventConditionsCheck(CreatureEventAI_Event const& event) const;
 
-        void DoFindFriendlyMissingBuff(CreatureList& list, float range, uint32 spellId) const;
```
Comment (killerwife) [inline src/game/AI/EventAI/CreatureEventAI.h:837]: Make one function with another parameter instead. Two grid searchers are correct however.

**Comment by killerwife** [comment]
@miraco and @AnonXS where is this needed?

**Comment by AnonXS** [comment]
supposedly this
``` 
('1659405','16594','16','0','100','1025','30479','40','10000','60000','0','0','11','30479','12','0','0','0','0','0','0','0','0','0','Shadowmoon Acolyte - Cast Resist Shadow on Friendly Missing Buff'),
```

should only or too be used OOC

**Comment by miraco** [comment]
It’s getting used in and out of combat

**Comment by AnonXS** [comment]
well then i mean we should still utilize Phases to use the same EVENT for Combat and NonCombat duplicating stuff only to check for combat, no combat doesnt make sense

**Comment by killerwife** [comment]
Well then, ACID has 6 Event parameters. Instead of adding "ooc" event, add a flag which governs whether its used in combat or out of combat or both :) A lot neater solution.

**Comment by miraco** [comment]
I updated it now. is this what you mean? 
using event_param5 as combat check

 // 0 = Only in combat
            // 1 = Out and in combat
            // 2 = Only out of combat

**Comment by miraco** [comment]
After finishing my work on shattered halls i wanted to continue with other instances so it would be nice to get a feedback for this
@killerwife

**Comment by killerwife** [comment]
Thanks for the reminder. I will assume its working so I am merging it.

**Comment by killerwife** [comment]
Oh, one more thing, please add documentation to EAI docs txt

**Comment by miraco** [comment]
Yes tested it again, still works. 
For EAI im gonna do a different Pr i guess

**Comment by killerwife** [comment]
I meant this file https://github.com/cmangos/mangos-tbc/blob/master/doc/EventAI.txt has to be updated to reflect the parameter addition.

**Pull request merged**